From c7956dec3803b7d0a84d1ceeaebcabe343318fa7 Mon Sep 17 00:00:00 2001
From: Luca Magrone <luca@magrone.cc>
Date: Fri, 27 Oct 2023 22:17:21 +0200
Subject: [PATCH] Add soversion to shared libraries

Follow the same versioning as the project for the shared libraries

Signed-off-by: Luca Magrone <luca@magrone.cc>
---
 src/Cedar/CMakeLists.txt   | 1 +
 src/Mayaqua/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/Cedar/CMakeLists.txt b/src/Cedar/CMakeLists.txt
index 7818d790..16b59208 100644
--- a/src/Cedar/CMakeLists.txt
+++ b/src/Cedar/CMakeLists.txt
@@ -17,6 +17,7 @@ set_target_properties(cedar
   ARCHIVE_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
   LIBRARY_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
   RUNTIME_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
+  SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
 )
 
 target_link_libraries(cedar PUBLIC mayaqua)
diff --git a/src/Mayaqua/CMakeLists.txt b/src/Mayaqua/CMakeLists.txt
index 32a121a9..49be4c3d 100644
--- a/src/Mayaqua/CMakeLists.txt
+++ b/src/Mayaqua/CMakeLists.txt
@@ -14,6 +14,7 @@ set_target_properties(mayaqua
   ARCHIVE_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
   LIBRARY_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
   RUNTIME_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
+  SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
 )
 
 find_package(OpenSSL REQUIRED)
-- 
2.39.3

